From: Prashant Bhole <bhole_prashant_q7@lab.ntt.co.jp>
Date: Tue, 13 Feb 2018 13:42:49 +0900
Subject: bpf: samples/sockmap fix Makefile for build error
Patch-mainline: v4.17-rc1
Git-commit: 9492686c53f3a98e7027d1079db1471ab20e17de
References: bsc#1109837

While building samples/sockmap, undefined reference error is thrown
for `nla_dump_errormsg'.
Linking tools/lib/bpf/nlattr.o as a fix

Signed-off-by: Prashant Bhole <bhole_prashant_q7@lab.ntt.co.jp>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 samples/sockmap/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/samples/sockmap/Makefile
+++ b/samples/sockmap/Makefile
@@ -5,7 +5,7 @@ obj- := dummy.o
 hostprogs-y := sockmap
 
 # Libbpf dependencies
-LIBBPF := ../../tools/lib/bpf/bpf.o
+LIBBPF := ../../tools/lib/bpf/bpf.o ../../tools/lib/bpf/nlattr.o
 
 HOSTCFLAGS += -I$(objtree)/usr/include
 HOSTCFLAGS += -I$(srctree)/tools/lib/
